tests/c/pthread_create.c

```diff
@@ -1,30 +1,42 @@
 // Compiler:
 // Run-time:
-//   status: error
-//   stderr: ...
-//    not yet implemented: Allocate and set thread-local ShadowStack instance. No support for threads yet!
-//    ...
+//   status: success
+//   stdout: 1,2,3,0
+//   stderr:
 
 #include <pthread.h>
+#include <stdio.h>
 #include <yk.h>
 #include <yk_testing.h>
 
-void *thread_function(void *arg) { return NULL; }
+void *thread_function(void *arg) {
+  int *a = (int *)arg;
+  printf("%d,", *a);
+  return NULL;
+}
 
 int main(int argc, char **argv) {
   YkMT *mt = yk_mt_new(NULL);
   yk_mt_hot_threshold_set(mt, 0);
   YkLocation loc = yk_location_new();
 
-  int i = 3;
-  while (i > 0) {
+  int MAX_THREADS = 3;
+  int thread_args[3] = {1, 2, 3};
+  pthread_t thread_handles[MAX_THREADS];
+
+  int a = 0;
```

Comment:
Also, try changing this to `int a = argc`. I'm not sure what optimisations we're currently running on top of `O0`, but one of them might statically work out that `a` is always 0 and remove the `if (a != 0)` case below entirely.

Reply:
ok, will give it a go!

Reply:
covered in 👉 ee405ad6580ec941fec1082b349e95ac2ba6fb81